guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java

```diff
@@ -82,8 +83,12 @@ public void registerFilters(List<Filter> filters, List<BindableService> services
                     });
                     // Key is of the form <Service Name>+ "/" +<Method Name>
                     // reflecting the structure followed in the gRPC HandlerRegistry using MethodDescriptor#getFullMethodName()
-                    filtersMap.put((service.bindService().getServiceDescriptor().getName() + "/" + pair.getValue().getName()).toLowerCase(),
-                            filtersForMethod);
+                    String methodSignature =
+                        (service.bindService().getServiceDescriptor().getName() + "/" + pair.getValue().getName()).toLowerCase();
+                    filtersMap.put(methodSignature, filtersForMethod);
+                    if (enableAccessLogs){
+                        filtersMap.get(methodSignature).add(new GrpcAccessLogFilter());
+                    }
```

Comment:
why not do this before, and get it added in `filtersForMethod` as required

Reply:
done